### DISCUSSION TITLE
How to use dot character for key when add new string

### DISCUSSION THREAD

**Discussion by amir135** [WeblateOrg/weblate#5675] [General] [answered]
Hello

I used n18next JSON format for front-end translation. when I have a problem in add a new string via Weblate.
My keys format is **"part1.paert2.part3"**. fro example.
`{
     "test.example.word":"Test Example Word"
}`
I import JSON files to Weblate successfully. 
But when I add a new string Weblate change my key to the below format:
`{
     "test":{
           "example":{
                 "word": "Test Example Word"
            }
       }
}`

Who can help me to add new strings that key contains the "." character?

**Comment by nijel** [comment, marked as answer]
For plain JSON files there are two format variants to toggle this behaviour, but it is currently hardcoded for i18next.

See also https://docs.weblate.org/en/latest/formats.html#json-files
